eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgChannelsDb.scala

```diff
@@ -238,6 +245,45 @@ class PgChannelsDb(implicit ds: DataSource, lock: PgLock) extends ChannelsDb wit
     }
   }
 
+  override def forgetHtlcInfos(channelId: ByteVector32, beforeCommitIndex: Long): Unit = withMetrics("channels/forget-htlc-infos", DbBackends.Postgres) {
+    withLock { pg =>
+      using(pg.prepareStatement("INSERT INTO local.htlc_infos_to_remove (channel_id, before_commitment_number) VALUES(?, ?) ON CONFLICT (channel_id) DO UPDATE SET before_commitment_number = EXCLUDED.before_commitment_number")) { statement =>
+        statement.setString(1, channelId.toHex)
+        statement.setLong(2, beforeCommitIndex)
+        statement.executeUpdate()
+      }
+    }
+  }
+
+  override def removeHtlcInfos(batchSize: Int): Unit = withMetrics("channels/remove-htlc-infos", DbBackends.Postgres) {
+    withLock { pg =>
+      // Check if there are channels that need to be cleaned up.
+      val channelToCleanUp_opt = using(pg.prepareStatement("SELECT channel_id, before_commitment_number FROM local.htlc_infos_to_remove LIMIT 1")) { statement =>
+        statement.executeQuery().map(rs => {
+          val channelId = ByteVector32(rs.getByteVector32FromHex("channel_id"))
+          val beforeCommitmentNumber = rs.getLong("before_commitment_number")
+          (channelId, beforeCommitmentNumber)
+        }).lastOption
+      }
+      // Remove a batch of HTLC information for that channel.
+      channelToCleanUp_opt.foreach { case (channelId, beforeCommitmentNumber) =>
+        val deletedCount = using(pg.prepareStatement(s"DELETE FROM local.htlc_infos WHERE channel_id=? AND commitment_number IN (SELECT commitment_number FROM local.htlc_infos WHERE channel_id=? AND commitment_number<? LIMIT $batchSize)")) { statement =>
+          statement.setString(1, channelId.toHex)
+          statement.setString(2, channelId.toHex)
+          statement.setLong(3, beforeCommitmentNumber)
+          statement.executeUpdate()
+        }
```

Comment:
This could be nice for debugging (otherwise I'll simply add it on a feature branch):
```suggestion
        }
        logger.info(s"deleted $deletedCount rows from htlc_infos for channelId=$channelId beforeCommitmentNumber=$beforeCommitmentNumber")
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2705/commits/ce507517f3cb112eea66e7143207686b92e90913